lib/ansible/plugins/lookup/cyberarkpassword.py

```diff
@@ -0,0 +1,142 @@
+                'AppDescs.AppID=' +
+                self.appid,
+                '-p',
+                'Query=' +
+                self.query,
+                '-o',
+                self.output,
+                '-d',
+                self.delimiter]
+            allParms.extend(self.extraParms)
+
+            credential = subprocess.check_output(allParms)
+
+            if len(credential) > 0 and credential.endswith("\n"):
+                credential = credential[:-1]
+
+            outputNames = self.output.split(",")
+            outputValues = credential.split(self.delimiter)
+
+            for i in range(len(outputNames)):
+                if outputNames[i].startswith("passprops."):
+                    if "passprops" not in resultDict:
+                        resultDict["passprops"] = {}
+                    outputPropName = outputNames[i][10:]
+                    resultDict["passprops"][outputPropName] = outputValues[i]
+                else:
+                    resultDict[outputNames[i]] = outputValues[i]
+
+        except subprocess.CalledProcessError as e:
+            raise AnsibleError(e.output)
+        except OSError as e:
+            raise AnsibleError("ERROR - AIM not installed or clipasswordsdk not in standard location. ERROR=(" + to_text(e.errno) + ") => " + e.strerror)
+
+        return [resultDict]
+
+
+class LookupModule(LookupBase):
+
+    """
+    USAGE: {{lookup('cyberarkpassword', appid='Application1', query='safe=Safe1;Folder=root;Object=User1',
+                    output='password,passprops.username,passprops.address' [, extraParms])}}
+
+    It Requires CyberArk AIM Installed, and /opt/CARKaim/sdk/clipasswordsdk in place or set environment variable AIM_CLIPASSWORDSDK_CMD to the AIM
+    CLI Password SDK executable.
+
+     Parameter names AppID, Query, Output are case sensitive.
+
+     Args:
+         appid (str): Defines the unique ID of the application that is issuing the password request.
+         query (str): Describes the filter criteria for the password retrieval.
+         output (str): Specifies the desired output fields separated by commas. They could be: Password, PassProps.<property>, PasswordChangeInProcess
+         Optionally, you can specify extra parameters recognized by clipasswordsdk (like FailRequestOnPasswordChange, Queryformat, Reason, etc.)
+
+     Returns:
+         dict: A dictionary with 'password' as key for the credential, passprops.<property>, passwordchangeinprocess
+               If the specified property does not exist for this password, the value <na> will be returned for this property.
+               If the value of the specified property is empty, <null> will be returned.
+
+
+    for extraParms values please check parameters for clipasswordsdk in CyberArk's "Credential Provider and ASCP Implementation Guide"
+
+    For Ansible on windows, please change the -parameters (-p, -d, and -o) to /parameters (/p, /d, and /o) and change the location of CLIPasswordSDK.exe
+    """
```

Comment:
In addition to this comment which explains the usage, you also need to add documentation in `docs/docsite/rst/playbooks_lookups.rst`